src/Type/MixedType.php

```diff
@@ -494,6 +497,32 @@ public function toFloat(): Type
 
 	public function toString(): Type
 	{
+		if ($this->subtractedType !== null) {
+			$castsToEmptyString = new UnionType([
+				new NullType(),
+				new ConstantBooleanType(false),
+				new ConstantStringType(''),
+			]);
+			if ($this->subtractedType->isSuperTypeOf($castsToEmptyString)->yes()) {
+				$accessories = [
+					new StringType(),
+					new AccessoryNonEmptyStringType(),
+				];
+
+				$castsToZeroString = new UnionType([
+					new ConstantFloatType(0.0),
+					new ConstantStringType('0'),
+					new ConstantIntegerType(0),
+				]);
+				if ($this->subtractedType->isSuperTypeOf($castsToZeroString)->yes()) {
+					$accessories[] = new AccessoryNonFalsyStringType();
+				}
+				return new IntersectionType(
+					$accessories,
```

Comment:
I think I'm missing floats here. See https://3v4l.org/7kgpL

Reply:
maybe I missed your point, but float `0.0` is substracted in line 544

Reply:
I was caught off guard a bit because this method (https://github.com/phpstan/phpstan-src/pull/3434/files#diff-23a89368c86f55bda583e138270384edec06835b889677bb17fc84e581e36dfcR494) mentions whole `FloatType` and this one mentions just constant `0.0`, is there a difference on purpose or does it even matter?

Reply:
the difference is, that there is a bunch of float numbers which cast to int `0`, but only a single one which casts to string `'0'`

https://3v4l.org/Q1cva

Reply:
Oh right :)